scanners/zap/zap.py

```diff
@@ -567,22 +570,28 @@ def _setup_passive_wait(self):
         self.automation_config["jobs"].append(waitfor)
 
     def _setup_active_scan(self):
-        """Adds the active scan job list."""
+        """Adds an active scan job list, if there is one"""
 
-        if self.my_conf("activeScan", default=False) is False:
+        params = self.config.subtree_to_dict(self.absolute_conf_path("activeScan"))
+        if params is None:
             return
 
-        active = {
+        job = {
             "name": "activeScan",
             "type": "activeScan",
-            "parameters": {
-                "context": Zap.DEFAULT_CONTEXT,
-                "user": Zap.USER if self.authenticated else "",
-                "policy": self.my_conf("activeScan.policy", default="API-scan-minimal"),
-            },
+            "parameters": params,
         }
 
-        self.automation_config["jobs"].append(active)
+        # Enforce some parameters
+        job["parameters"]["user"] = Zap.USER if self.authenticated else ""
+        job["parameters"]["context"] = Zap.DEFAULT_CONTEXT
```

Comment:
Can we create a new function to centralize this block of L579 (`job = {`) to L587 (adding 'user' and 'context' parameters)? Because this block seems to have a chance to become redundant for most setup_*().

Reply:
updated: created a `self._enforce_job_parameters` function that does that. 
"spider" job also accepts both context/user parameters, so I converted `_setup_spider` to this new method.

now `spider`, `ajaxSpider`, and `openapi` all 3 uses `self._enforce_job_parameters`